packages/edit-widgets/src/blocks/widget-area/edit/index.js

```diff
@@ -24,6 +30,11 @@ export default function WidgetAreaEdit( {
 	className,
 	attributes: { id, name },
 } ) {
+	const areaClassName = classnames(
+		className,
+		'widget-area',
+		cleanForSlug( name )
```

Comment:
Would it be possible to use the `id` provided to `register_sidebar()` instead? The `name` field is meant to be displayed to the user, so it's likely to be longer and potentially containing special characters. This would make predicting what the class will be after cleaning a bit of a guessing game.

The ID is meant to be the machine readable identifier, so I think it would be simpler to use.

Reply:
I don't think we should just use the `id` verbatim either, we should give it a prefix/postfix to avoid the chance of it clashing with other class names that might be in a stylesheet.

e.g. `widget-area-${ id }` or `is-${ id }-widget-area`.

Reply:
I'm not sure I follow why we need this PR?

Is this for third-party developers to style them individually? If that's the case then technically these class names are now considered public API? We'll probably have to document this somewhere, comments at the very least, to prevent contributors accidentally removing or changing them.

I don't think we should use class names for them though. In my mind, class names should be considered private API (I know it's not the case now though 😅) and are only used for styling in this plugin or core. Instead, I think we should use `data-*` attribute for third-party developers and for public APIs. I think we already have `data-type="core/widget-area"`, maybe we could also add something like `data-widget-area-id` and something similar for previews. WDYT?

Reply:
This is not a bad suggestion @kevin940726. @desrosj this would accomplish our goal, but without fiddling with editor's CSS.

Reply:
That would also accomplish the goal of this, so I'm OK with this approach as well. `data-widget-area-id` also feels good to me.